lib/elixir/lib/registry.ex

```diff
@@ -171,6 +171,15 @@ defmodule Registry do
   @typedoc "The type of registry metadata values"
   @type meta_value :: term
 
+  @typedoc "A pattern to match on objects in a registry"
+  @type match_pattern :: atom | term
+
```

Comment:
Here it should be `atom | tuple`. But we could be more detailed if tuple follows a specific format

Reply:
The pattern can be any term.

Reply:
Then the documentation should be corrected accordingly. Now it says:
> Pattern must be an atom or a tuple that will match the structure of the value stored in the registry.

Reply:
I think the documentation is wrong. Since the client code is what ultimately defines what goes inside the registry as value for a key, I should probably change the type to be just `term` and update the documentation. Thoughts?

Reply:
I was thinking about updating the documentation to read:

> Pattern must be any term that will match the structure of the value stored in the registry.